packages/main/src/plugin/provider-registry.ts

```diff
@@ -604,12 +604,14 @@ export class ProviderRegistry {
     }
 
     const provider = this.getMatchingProvider(providerInternalId);
-    let connection: ContainerProviderConnection | KubernetesProviderConnection | undefined;
+    let connection: ContainerProviderConnection | KubernetesProviderConnection | VmProviderConnection | undefined;
```

Comment:
not for this PR but like follow-up: add a union type for `ContainerProviderConnection | KubernetesProviderConnection | VmProviderConnection`

Reply:
for the moment `ContainerProviderConnection` and `KubernetesProviderConnection` are defined in `extension-api.d.ts` but `VmProviderConnection` is still defined internally (before https://github.com/podman-desktop/podman-desktop/issues/11748). The union type will be defined when we move the definition of `VmProviderConnection` in `extension-api.d.ts` (cf @SoniaSandler)